compiler/noirc_evaluator/src/ssa/opt/inlining.rs

```diff
@@ -1150,4 +1207,171 @@ mod test {
         let ssa = ssa.inline_functions(i64::MIN).unwrap();
         assert_normalized_ssa_equals(ssa, no_inline_always_src);
     }
+
+    #[test]
+    fn acir_global_arrays_are_inlined_with_new_value_ids() {
+        let src = "
+        g0 = Field 1
+        g1 = Field 2
+        g2 = make_array [Field 1, Field 2] : [Field; 2]
+
+        acir(inline) fn main f0 {
+          b0():
+            v0 = call f1() -> [Field; 2]
+            return v0
+        }
+        acir(inline) fn create_array f1 {
+          b0():
+            return g2
+        }
+        ";
+
+        let ssa = Ssa::from_str(src).unwrap();
+        let ssa = ssa.inline_functions(i64::MAX).unwrap();
+
+        assert_ssa_snapshot!(ssa, @r"
+        g0 = Field 1
+        g1 = Field 2
+        g2 = make_array [Field 1, Field 2] : [Field; 2]
+
+        acir(inline) fn main f0 {
+          b0():
+            v3 = make_array [Field 1, Field 2] : [Field; 2]
+            return v3
+        }
+        ");
+    }
+
+    #[test]
+    fn brillig_global_arrays_keep_same_value_ids() {
+        let src = "
+        g0 = Field 1
+        g1 = Field 2
+        g2 = make_array [Field 1, Field 2] : [Field; 2]
+
+        brillig(inline) fn main f0 {
+          b0():
+            v0 = call f1() -> [Field; 2]
+            // v1 = array_get g2, index u32 1 -> Field
```

Comment:
Ah, I forgot to point out this comment too, sorry.

Reply:
Oops I missed that. We can just remove in some other testing PR